actor/src/main/scala-3/org/apache/pekko/util/ByteIterator.scala

```diff
@@ -482,7 +482,7 @@ abstract class ByteIterator extends BufferedIterator[Byte] {
 
   override def indexWhere(p: Byte => Boolean, from: Int = 0): Int = {
     var index = 0
-    while (index < from) {
+    while (index < from && hasNext) {
```

Comment:
the changes in byte Iterator is not needed.

IN Netty, these method is named as firstIndexOf and lastIndexOf, that's an option too.

Reply:
Extracted to https://github.com/apache/pekko/pull/1282